OpenRA.Mods.Common/Orders/PlaceBuildingOrderGenerator.cs

```diff
@@ -154,7 +154,7 @@ CPos TopLeft
 			{
 				var offsetPos = Viewport.LastMousePos;
 				if (variants[variant].Preview != null)
-					offsetPos += variants[variant].Preview.TopLeftScreenOffset;
+					offsetPos = viewport.WorldToViewPx(viewport.ViewToWorldPx(offsetPos) + variants[variant].Preview.TopLeftScreenOffset);
```

Comment:
```suggestion
					offsetPos += viewport.WorldToViewPx(variants[variant].Preview.TopLeftScreenOffset);
```

why not?

Reply:
`WorldToViewPx` applies to positions, and gives the wrong result when applied to vectors (it isn't just a multiplication, it also changes the origin).

Reply:
could we do some simplification here anyway? The ViewToWorld function doesn't even want ViewPx, it immediatelly turns it back to WorldPx. If feels we are converting from and to for no reason

Reply:
`ViewToWorld` converts from viewport pixels to CPos, and works very differently from `ViewToWorldPx`! I don't see any straightforward simplifications that can be done.